docs/source/delta-spark-connect.md

```diff
@@ -0,0 +1,121 @@
+2. Simpler integration of Spark and <Delta> with developer tooling. IDEs no longer have to integrate with the full Spark and <Delta> implementation, and instead can integrate with a thin-client.
+3. Support for languages other than Java/Scala and Python. Clients "merely" have to generate Protocol Buffers and therefore become simpler to implement.
+4. Spark and <Delta> will become more stable, as user code is no longer running in the same JVM as Spark's driver.
+5. Remote connectivity. Code can run anywhere now, as there is a gRPC layer between the user interface and the driver.
+
+## Supported Delta Connect operations
+
+The feature introduces a limited set of supported operations in <Delta> 4.0 preview and expands it in <Delta> 4.0 and above.
+
+.. csv-table::
+:header: "Supported operations - Delta 4.0 preview", "Supported operations - Delta 4.0 release"
+
+"`forPath`","`forPath`"
+"`forName`","`forName`"
+"`alias`","`alias`"
+"`toDF`","`toDF`"
+,"`generate`"
+,"`vacuum`"
+,"`delete`"
+,"`update`"
+,"`merge`"
+,"`history`"
+,"`detail`"
+,"`convertToDelta`"
+,"`create`"
+,"`createIfNotExists`"
+,"`replace`"
+,"`createOrReplace`"
+,"`isDeltaTable`"
+,"`upgradeTableProtocol`"
+,"`restoreToVersion`"
+,"`restoreToTimestamp`"
+,"`optimize`"
+,"`clone`"
+
+## How to start the Spark Server with Delta
+
+1. Download `spark-4.0.0-preview1-bin-hadoop3.tgz` from [Spark 4.0.0 First Preview](https://archive.apache.org/dist/spark/spark-4.0.0-preview1).
+
+2. Start the Spark Connect server with the <Delta> Connect plugins:
+
+```bash
+sbin/start-connect-server.sh \ 
+  --packages org.apache.spark:spark-connect_2.13:4.0.0-preview1,io.delta:delta-connect-server_2.13:4.0.0-preview1,io.delta:delta-connect-common_2.13:4.0.0-preview1,com.google.protobuf:protobuf-java:3.25.1 \ 
+  --conf "spark.sql.extensions=io.delta.sql.DeltaSparkSessionExtension" \
+  --conf "spark.sql.catalog.spark_catalog=org.apache.spark.sql.delta.catalog.DeltaCatalog" \
+  --conf "spark.connect.extensions.relation.classes=org.apache.spark.sql.connect.delta.DeltaRelationPlugin" \
+  --conf "spark.connect.extensions.command.classes=org.apache.spark.sql.connect.delta.DeltaCommandPlugin"
+```
+
+## How to use the Python Spark Connect Client with Delta
+
+The <Delta> Connect Python client is included in the same PyPi package as <Delta> Spark.
+
+1. `pip install pyspark==4.0.0`
+
+2. `pip install delta-spark==4.0.0`
+
+3. There is no difference in usage compared to the classic way. We just need to pass in a remote `SparkSession` (instead of a local one) to the `DeltaTable` API.
+
+An example:
+
+```python
+import io.delta.tables._
+import org.apache.spark.sql.functions._
+
+val deltaTable = DeltaTable.forName(spark, "my_table")
+
+deltaTable.update(
+  condition = expr("id % 2 == 0"),
+  set = Map("id" -> expr("id + 100")))
+```
+
+## How to use the Scala Spark Connect Client with Delta
```

Comment:
Add a note, saying these set of steps are going to be simplified with final Spark 4.0 and Delta 4.0